UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs

```diff
@@ -1670,6 +1671,26 @@ private void LayerCanvas_Unloaded(object sender, RoutedEventArgs e)
             if (canvas.CurrentLayer is not null)
                 ObjElemDict.Remove(canvas.CurrentLayer);
         }
+        private void SetupRoomWithGrids(UndertaleRoom room)
+        {
+            if (Settings.Instance.GridWidthEnabled)
+            {
+                room.GridWidth = Settings.Instance.GlobalGridWidth;
+                if (Settings.Instance.GridHeightEnabled)
+                {
+                    room.GridHeight = Settings.Instance.GlobalGridHeight;
+                    room.SetupRoom(false, false);
+                }
+                else
+                {
+                    room.SetupRoom(false);
+                }
+            }
+            else
+            {
+                room.SetupRoom();
+            }
```

Comment:
This logic here is very weird.
The way it is coded right now, it will only use a forced height, if a forced width is given as well.